submissions/tavor118sn/task_friends_app/script.js

```diff
@@ -0,0 +1,156 @@
+  userImg.classList.add('user-photo');
+  userWholeCard.appendChild(userImg);
+
+  const cardHeader = document.createElement('h2');
+  cardHeader.classList.add('user-full-name');
+  cardHeader.innerHTML = `${userInfo.name.first} ${userInfo.name.last}`;
+  userWholeCard.appendChild(cardHeader);
+
+  const genderElem = document.createElement('p');
+  genderElem.innerHTML = `Gender: <strong>${userInfo.gender}</strong>`;
+  userWholeCard.appendChild(genderElem);
+
+  const ageElem = document.createElement('p');
+  ageElem.innerHTML = `Age: <strong>${userInfo.dob.age}</strong>`;
+  ageElem.classList.add('user-age');
+  userWholeCard.appendChild(ageElem);
+
+  return userWholeCard;
+};
+
+const generateFriendsListing = (friendList) => {
+  CARD_LIST.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+
+  friendList.forEach((elem) => {
+    fragment.appendChild(createCard(elem));
+  });
+
+  CARD_LIST.appendChild(fragment);
+};
+
+// Load users
+
+async function loadUserList() {
+  const newUrl = buildUrl();
+  // try to make request at least 5 times in case of errors
+  for (let i = 0; i < 5; i++) {
+    try {
+      const response = await fetch(newUrl);
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      } else {
+        const friends = await response.json();
+        generateFriendsListing(friends.results);
+        return friends.results;
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  }
+}
+
+// Sorting and filtering
+
+const getFormData = (form) => {
+  return {
+    userSortAge: form.userSortAge.value,
+    userSortName: form.userSortName.value,
+    userFilterGender: form.userFilterGender.value,
+  };
+};
+
+const filterByGender = (gender) => {
+  let friendList = [...friends];
+  if (gender !== 'all') {
+    friendList = friendList.filter((user) => user.gender === gender);
+  }
+  return friendList;
+};
+
+const filterByName = (event) => {
+  const enteredFirstName = event.target.value;
+
+  let friendList = [...friends];
+  friendList = friendList.filter(
+    (user) => user.name.first.toLowerCase().search(enteredFirstName.toLowerCase()) !== -1
+  );
+
+  generateFriendsListing(friendList);
+};
+
+const sortAndFilterListing = (event) => {
+  event.preventDefault();
+  const formData = getFormData(FORM);
+
+  const sortFunc = (a, b) => {
+    let sortAge = 0;
+    if (formData.userSortAge === ASCENDING_ORDER) {
+      sortAge = a.dob.age - b.dob.age;
+    } else if (formData.userSortAge === DESCENDING_ORDER) {
+      sortAge = b.dob.age - a.dob.age;
+    }
+
+    const nameA = a.name.last.toUpperCase();
+    const nameB = b.name.last.toUpperCase();
+    let sortName;
+    if (formData.userSortName === ASCENDING_ORDER) {
+      sortName = nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    } else if (formData.userSortName === DESCENDING_ORDER) {
+      sortName = nameA < nameB ? 1 : nameA > nameB ? -1 : 0;
+    }
+    return sortAge || sortName;
```

Comment:
Sorting by age and by name separate functions ;)

Reply:
@zonzujiro In my implementation it can be used together (Sorting by age and by name) with sortAge priority.
Should I rewrite it?
(to use only one sorting at the time)

Reply:
You can't sort a list of items by two conditions simultaneously. It's impossible.

Reply:
Okay, got it! I've just used this response from stackoverflow
https://stackoverflow.com/questions/6129952/javascript-sort-array-by-two-fields

Reply:
First of all, what you did called stackoverflow development.

Second - you can sort by age OR by name.

> Sorting by age and by name

So this is not right.

Create separate functions for sorting and then call them depending on what sorting condition was selected.

Reply:
Fixed. I hope I undersood corectly